SmartDeviceLink/SDLTouchManager.m

```diff
@@ -257,16 +276,80 @@ - (void)sdl_handleTouchEnded:(SDLTouch *)touch {
                 self.singleTapTouch = nil;
             }
             break;
-        case SDLPerformingTouchTypeNone:
+
+            case SDLPerformingTouchTypeNone:
             break;
     }
     self.previousTouch = nil;
     _performingTouchType = SDLPerformingTouchTypeNone;
 }
 
+/**
+ *  Handles a CANCEL touch event sent by CORE. The CANCEL touch event is sent when a gesture is interrupted during a video stream. This can happen when a system dialog box appears on the screen, such as when the user is alerted about an incoming phone call. 
+ *
+ *  Pinch and pan gesture subscribers are notified if the gesture is canceled. Tap gestures are simply canceled without notification.
+ *
+ *  @param touch    Gesture information
+ */
+- (void)sdl_handleTouchCanceled:(SDLTouch *)touch {
+    if (!self.isTouchEnabled) {
+        return; // no-op
+    }
+
+    if (self.singleTapTimer != nil) {
+        // Cancel any ongoing single tap timer
+        [self sdl_cancelSingleTapTimer];
+        self.singleTapTouch = nil;
+    }
+
+    switch (self.performingTouchType) {
+            case SDLPerformingTouchTypeMultiTouch:
+            [self sdl_setMultiTouchFingerTouchForTouch:touch];
+            if (self.currentPinchGesture.isValid) {
+                if ([self.touchEventDelegate respondsToSelector:@selector(touchManager:pinchCanceledAtCenterPoint:)]) {
+                    [self.touchEventDelegate touchManager:self
+                                 pinchCanceledAtCenterPoint:self.currentPinchGesture.center];
+                }
+                self.currentPinchGesture = nil;
+            }
+            break;
+
+            case SDLPerformingTouchTypePanningTouch:
+            if ([self.touchEventDelegate respondsToSelector:@selector(touchManager:panningCanceledAtPoint:)]) {
+                [self.touchEventDelegate touchManager:self
+                                 panningCanceledAtPoint:touch.location];
+            }
+            break;
+
+            case SDLPerformingTouchTypeNone:
+            case SDLPerformingTouchTypeSingleTouch:
```

Comment:
Reverse this and the above case, they should be in the same order as other sub-handlers (this is done in the code provided in the above comment)

Reply:
Fixed